master/buildbot/changes/gitpoller.py

```diff
@@ -63,7 +67,8 @@ def __init__(self, repourl, branches=None, branch=None,
 
         base.PollingChangeSource.__init__(self, name=name,
                                           pollInterval=pollInterval,
-                                          pollAtLaunch=pollAtLaunch)
+                                          pollAtLaunch=pollAtLaunch,
+                                          sshPrivateKey=sshPrivateKey)
```

Comment:
why do you need to pass it to parent if your store it in this class?

Reply:
It turns out that the BuildbotService requires any secrets to be rendered to be passed as kwargs to __init__. See https://github.com/buildbot/buildbot/blob/dfad61a971c696a4688181a06ae139241799deda/master/buildbot/util/service.py#L211.